From edb907508d27d6f63e101bab8d34740fc4dc67d5 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Tue, 3 Nov 2015 19:11:18 -0800
Subject: [PATCH 0328/1239] serial: Wait for FIFO to empty when transferring
 characters

This ensures that all characters are sent on the wire.
---
 drivers/serial/serial_pl01x.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/serial/serial_pl01x.c b/drivers/serial/serial_pl01x.c
index 2a5f256184..92c85b2da6 100644
--- a/drivers/serial/serial_pl01x.c
+++ b/drivers/serial/serial_pl01x.c
@@ -40,6 +40,9 @@ static int pl01x_putc(struct pl01x_regs *regs, char c)
 	/* Send the character */
 	writel(c, &regs->dr);
 
+	while (!(readl(&regs->fr) & UART_PL01x_FR_TXFE))
+		WATCHDOG_RESET();
+
 	return 0;
 }
 
-- 
2.29.0

